Intersections_2/include/CGAL/Intersections_2/internal/Triangle_2_Triangle_2_intersection_impl.h

```diff
@@ -308,35 +319,68 @@ Triangle_2_Triangle_2_pair<K>::intersection_point() const
 }
 
 
+//algorithm taken from here : https://stackoverflow.com/questions/1165647/how-to-determine-if-a-list-of-polygon-points-are-in-clockwise-order
+template <typename K,  typename Exact>
+struct Is_cw
+{
+  bool operator()(const std::vector<typename K::Point_2>& ps)
+  {
+    typename K::FT res(0);
+    std::size_t length = ps.size();
+    for(std::size_t i = 0; i<length; ++i)
+      res += (ps[(i+1)%length].x() - ps[i].x())*(ps[(i+1)%length].y()+ps[i].y());
+
+    return res > 0;
+  }
+};
+
+template <typename K>
+struct Is_cw<K, Tag_true>
+{
+  bool operator()(const std::vector<typename K::Point_2>& ps)
+  {
+    return !CGAL::left_turn(ps[0], ps[1], ps[2]);
```

Comment:
I'd say that return !left_turn allows points to be aligned, and that's why not simply right_turn()...

Reply:
If points were allowed to be aligned, your code would not be correct since you could then not just look at the first three points to deduce whether it's cw or ccw (because they could be aligned and then you would not be able to conclude).